tests/test_crud_immunisation_api.py

```diff
@@ -42,23 +42,37 @@ def test_crud_immunization_nhs_login(nhsd_apim_proxy_url, nhsd_apim_auth_headers
 
     # CREATE
     result = imms_api.create_immunization(imms)
-    res_body = result.json()
 
+    assert result.text == ""
     assert result.status_code == 201
-    assert res_body["resourceType"] == "Immunization"
+    assert "Location" in result.headers
 
     # READ
-    imms_id = res_body["id"]
+    imms_id = parse_location(result.headers["Location"])
 
     result = imms_api.get_immunization_by_id(imms_id)
+    res_body = result.json()
 
     assert result.status_code == 200
     assert res_body["id"] == imms_id
 
+    # UPDATE
+    new_imms = copy.deepcopy(imms)
+    new_imms["id"] = imms_id
```

Comment:
Should probably be `updated_imms` as it's not new if it's for the same id. A new imms would get a 201.

Reply:
It is "new" because it's the result of the previous step. But I agree it's confusing. I can't rename it to `updated` because it's not updated yet either lol.
I think I rename it to `update_payload`. This way, we know it's the imms request body that's about to get updated.